From fa6eb0d55a729cd63faca1b00228021ba1ca78fb Mon Sep 17 00:00:00 2001
From: Alexandre Torgue <alexandre.torgue@st.com>
Date: Wed, 23 Sep 2020 12:14:03 +0200
Subject: [PATCH 092/529] ARM: dts: stm32: Add power domain on stm32mp151

Enable stm32-pm-domain driver and add two core power domains:
core and core_ret power domains.

Signed-off-by: Alexandre Torgue <alexandre.torgue@st.com>
Change-Id: Ifa591d7d6d914372e4bc36feee1f3bf9cf2040f1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/178022
Reviewed-by: CITOOLS <smet-aci-reviews@lists.codex.cro.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/276993
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Tested-by: Pascal PAILLET <p.paillet@st.com>
Domain-Review: Alexandre TORGUE <alexandre.torgue@foss.st.com>
---
 arch/arm/boot/dts/stm32mp151.dtsi | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

--- a/arch/arm/boot/dts/stm32mp151.dtsi
+++ b/arch/arm/boot/dts/stm32mp151.dtsi
@@ -110,6 +110,26 @@
 		status = "disabled";
 	};
 
+	pm_domain {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "st,stm32mp157c-pd";
+
+		pd_core_ret: core-ret-power-domain@1 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <1>;
+			#power-domain-cells = <0>;
+			label = "CORE-RETENTION";
+
+			pd_core: core-power-domain@2 {
+				reg = <2>;
+				#power-domain-cells = <0>;
+				label = "CORE";
+			};
+		};
+	};
+
 	soc {
 		compatible = "simple-bus";
 		#address-cells = <1>;
